### ISSUE TITLE
US Gov Cloud Error: updating Recovery Service Storage Cfg Vault

### ISSUE THREAD

**Issue by 2good4hisowngood** [hashicorp/terraform-provider-azurerm#15066]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform version: 1.1.3
AzureRM version: 2.93.0

### Affected Resource(s)

* `azurerm_recovery_services_vaultc`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "2.93.0"
    }
  }
}
provider "azurerm" {
  subscription_id = "redacted"
  client_id       = "redacted"
  client_secret   = "redacted"
  tenant_id       = "redacted"
  environment     = "usgovernment"
  features {}
}
resource "azurerm_resource_group" "rg1" {
  name     = "test-prod-rg1"
  location = "US Gov Arizona"
}
resource "azurerm_recovery_services_vault" "rv1" {
  name                = "test-recovery-vault1"
  location            = azurerm_resource_group.rg1.location
  resource_group_name = azurerm_resource_group.rg1.name
  sku                 = "Standard"
}

```

### Expected Behaviour

azurerm_resource_group.rg1: Creation complete after 1s [id=/subscriptions/"redacted"/resourceGroups/test-prod-rg1]
azurerm_recovery_services_vault.rv1: Creation complete after Xs [resourceID]

### Actual Behaviour

azurerm_resource_group.rg1: Creation complete after 1s [id=/subscriptions/"redacted"/resourceGroups/test-prod-rg1]
Error: updating Recovery Service Storage Cfg Vault: (Name "test-recovery-vault1" / Resource Group "test-prod-rg1"): backup.ResourceStorageConfigsClient#Update: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="SubscriptionNotFound" Message="The subscription 'redacted' could not be found."

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This is running in US Gov Cloud

### References

* #0000

**Comment by 2good4hisowngood** [comment]
This issue started within the past week (1/15 first occurrence), and previously was working. I've verified it with 2.92.0 as well.

If you try to run it a second time it will tell you that the vault already exists and needs to be imported. 

Error: A resource with the ID "/subscriptions/'redacted'/resourceGroups/test-prod-rg1/providers/Microsoft.RecoveryServices/vaults/test-recovery-vault1" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_recovery_services_vault" for more information.

**Comment by sinbai** [comment]
@2good4hisowngood thanks for reporting this issue. It would be fixed when PR #15316  is released.

**Comment by tombuildsstuff** [comment]
Fixed via https://github.com/hashicorp/terraform-provider-azurerm/pull/15316
